forge-game/src/main/java/forge/game/card/CardState.java

```diff
@@ -377,6 +378,15 @@ public final SpellAbility getFirstSpellAbility() {
         return Iterables.getFirst(getNonManaAbilities(), null);
     }
 
+    public final SpellAbility getFirstSpellAbilityWithFallback() {
+        SpellAbility sa = getFirstSpellAbility();
+        if (sa != null) {
+            return sa;
+        }
+        // this happens if it's transformed backside (e.g. Disturbed)
+        return new SpellPermanent(getCard());
```

Comment:
What happens if the backside is a Land? Then it shouldn't make SpellPermament?

Reply:
Hmn, I can't think of how this would happen currently since this will only get called if it's on the Stack already

but I can add a safety check nonetheless

Reply:
You probably need to add `this` as second Param for the SpellPermanent for the CardState?
Otherwise, it probably would enter as wrong CardState?

Also, you might update Disturb Keyword to use this new Function?

Reply:
Fixed but I'm not sure how Disturb should use it

Reply:
For Disturb, shouldn't this be enough?

```java
SpellAbility newSA = host.getAlternateState().getFirstSpellAbilityWithFallback().copyWithDefinedCost(disturbCost);
```

Reply:
ah right, works fine

Reply:
I guess it does copy the SpellPermanent now instead, but shouldn't hurt too much?